utils/common/src/main/java/org/apache/brooklyn/util/javalang/Reflections.java

```diff
@@ -242,7 +242,13 @@ public Object loadInstance(String classname) throws ReflectionNotFoundException,
 
     /** Invokes a suitable constructor, supporting varargs and primitives */
     public static <T> Optional<T> invokeConstructorWithArgs(Class<T> clazz, Object[] argsArray, boolean setAccessible) {
-        Reflections reflections = new Reflections(clazz.getClassLoader());
+        ClassLoader cl = clazz.getClassLoader();
```

Comment:
It's very specific to this method call as we don't really care what the fallback classloader is.